rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYaml.java

```diff
@@ -37,23 +45,23 @@
             example = "labels: \n\tlabel-one: \"value-one\"")
     @Language("yml")
     String yaml;
+    @Option(displayName = "Filter YAML snippet",
+            description = "The YAML snippet to to use as a basis for filtering which files are applicable to this recipe.",
+            example = "labels: \n\tlabel-two: \"value-two\"",
+            required = false)
+    @Language("yml")
+    @Nullable
+    String filterYaml;
 
     @Override
     public Validated validate() {
-        return super.validate()
-                .and(Validated.test("yaml", "Must be valid YAML",
-                        yaml, y -> {
-                            List<Yaml.Documents> docs = new YamlParser().parse(yaml);
-                            if (docs.size() == 0) {
-                                return false;
-                            }
-                            Yaml.Documents doc = docs.get(0);
-                            if (doc.getDocuments().size() == 0) {
-                                return false;
-                            }
-                            Yaml.Block block = doc.getDocuments().get(0).getBlock();
-                            return block instanceof Yaml.Mapping;
-                        }));
+        Validated v = super.validate();
+        if (!StringUtils.isNullOrEmpty(filterYaml)) {
```

Comment:
This would be a more useful validation if what it asserted was that `new YamlParser().parse(filterYaml).get(0);` is not empty/invalid. You might also want to validate that `filterYaml` has only one root element, since if someone has two they're probably making a mistake about how this visitor works and will be disappointed/confused by its output.